lib/InstabugNavigatorObserver.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:instabug_flutter/Instabug.dart';
+
+class InstabugNavigatorObserver extends NavigatorObserver {
+
+  final List<Route> _steps = <Route>[];
+  
+  void screenChanged(final Route newRoute) {
+    // If there is a step that hasn't been pushed yet
+    if (_steps.isNotEmpty) {
+        // Report the last step and remove it from the list
+        Instabug.reportScreenChange(_steps[_steps.length-1].settings.name.toString());
+       _steps.remove(_steps[_steps.length-1]);
```

Comment:
We have two threads that manipulate the steps list. My concern is that this could potentially throw an out of bounds exception when dart swaps async operations between lines (12, 13) and line 21.

A few suggestions to try:
- Reference the `_steps[_steps.length - 1]` before executing any API, we don't know when the other thread will remove the step from the list but the sooner we reference it the better.
- [at L13 and at L21] Reorder the lines so that `_steps.remove` gets executed before `Instabug.reportScreenChange` and the steps length value is reflected right away. That'll lower the probability of failure; it won't prevent it though.
- Try & catch (the whole method?)